openhtf/util/logs.py

```diff
@@ -199,40 +201,48 @@ def emit(self, record):
     self._notify_update()
 
 
+class CliFormatter(logging.Formatter):
+  """Formats log messages for printing to the CLI."""
+  def format(self, record):
+    """Format the record as tersely as possible but preserve info."""
+    super(CliFormatter, self).format(record)
+    localized_time = datetime.datetime.fromtimestamp(record.created)
+    terse_time = localized_time.strftime(u'%H:%M:%S')
+    terse_level = record.levelname[0]
+    terse_name = record.name.split('.')[-1]
+    if record.name.startswith(RECORD_LOGGER_PREFIX):
+      # The [1:] split used here is to discard the extra leading dot.
+      loggers = record.name.split(RECORD_LOGGER_PREFIX)[1][1:].split('.')
+      # If nested under a test record, show where the message came from.
+      if len(loggers) > 1:
```

Comment:
this is making some assumptions about how users have set up their logger hierarchy, might want to explicitly check for 'plug' and 'phase'?

Reply:
Okay, cleaned this up a bunch using regexes too.